lib/connectors/mongodb/connector.rb

```diff
@@ -56,30 +70,166 @@ def initialize(configuration: {})
         @direct_connection = configuration.dig(:direct_connection, :value)
       end
 
-      def yield_documents
+      def yield_documents(job_description = {}, &on_doc_serialization)
+        advanced_filter_config = extract_filter_config(job_description)
+        rules = extract_rules(job_description)
+
+        check_filter_config!(rules, advanced_filter_config)
+
+        function, args = setup_db_function_with_args(advanced_filter_config)
+
+        call_db_function_on_collection(function, args, &on_doc_serialization)
+      end
+
+      private
+
+      def extract_rules(job_description)
+        Utility::Commons.return_if_present(job_description.dig(:filtering, :rules), DEFAULT_RULES)
+      end
+
+      def extract_filter_config(job_description)
+        Utility::Commons.return_if_present(job_description.dig(:filtering, :advanced_config), DEFAULT_ADVANCED_CONFIG)
+      end
+
+      def setup_db_function_with_args(filter_config)
+        # default parameters
+        function = FIND
+        args = NO_ARGS
+
+        function, args = setup_find(filter_config) if find_present?(filter_config)
+
+        function, args = setup_aggregate(filter_config) if aggregate_present?(filter_config)
+
+        [function, args]
+      end
+
+      def check_filter_config!(rules, config)
+        return unless filtering_present?(rules, config)
+
+        check_find_and_aggregate_present!(config)
+
+        check_find_and_aggregate_missing!(config)
```

Comment:
These two methods can actually be replaced by a single simpler method.

Reply:
This method can just verify 2 things:
1. Only one hash key is present
2. This key's value is expected ('find' or 'aggregate')